src/main/webapp/app/exam/participate/exam-participation.service.ts

```diff
@@ -246,4 +246,42 @@ export class ExamParticipationService {
             });
         }
     }
+
+    public static getSubmissionForExercise(exercise: Exercise) {
+        if (
+            exercise &&
+            exercise.studentParticipations &&
+            exercise.studentParticipations.length > 0 &&
+            exercise.studentParticipations[0].submissions &&
+            exercise.studentParticipations[0].submissions.length > 0
+        ) {
+            return exercise.studentParticipations[0].submissions[0];
```

Comment:
```suggestion
            exercise &&
            exercise.studentParticipations &&
            exercise.studentParticipations.length > 0 &&
            exercise.studentParticipations[0].submissions
        ) {
            return exercise.studentParticipations[0].submissions[0];
```

Reply:
As `exercise.studentParticipations[0].submissions` is present, `exercise.studentParticipations[0].submissions[0]` while having an empty array should return undefined, or am I wrong?